src/common/src/types/interval.rs

```diff
@@ -81,6 +81,59 @@ impl Interval {
         self.usecs.rem_euclid(USECS_PER_DAY) as u64
     }
 
+    /// Returns the years field.
+    pub fn years(&self) -> i32 {
+        self.months / 12
+    }
+
+    /// Returns the months field. range: 0-11
+    pub fn months(&self) -> i32 {
+        self.months % 12
+    }
+
+    /// Returns the days field.
+    pub fn days(&self) -> i32 {
+        self.days
+    }
```

Comment:
Maybe implement these helpers on a newtype? We already have `get_months` and `get_days` to get the individual fields and this results in:
* `get_months() != months()`
* `get_days() == days()`

Other ideas of differentiating the 2 versions of `months` functions are also welcomed.

Reply:
Maybe we can follow the naming convention in chrono. Use [`num_units`](https://docs.rs/chrono/latest/chrono/struct.Duration.html#method.num_weeks) for total number of the unit, and `units` for the number of the field?

Reply:
Not sure if I get it, does this convention mean `num_months` could return a value >= 12 and we need to rename existing `get_months` to it? And chrono does not have `units` for returning 0..=11 at the same time.

The idea of having 2 types is similar to `Interval` vs `pg_itm` [here](https://github.com/postgres/postgres/blob/REL_15_2/src/include/datatype/timestamp.h#L53).

Reply:
Yes.

I still prefer introducing different functions in a single type, since converting to a new type like `pg_itm` may cause unnecessary calculations.

Reply:
> converting to a new type like `pg_itm` may cause unnecessary calculations.

So I am not saying to convert to a new type. I meant using the rust `newtype` pattern. Calculation is still on demand.

```
struct Itm(Interval); // need a better name

impl Itm {
  pub months(&self) -> i32 {
    self.0.months % 12
  }
}

// on use: Itm(v).months()
```

Reply:
I see. This looks good.